ra2ce/graph/networks.py

```diff
@@ -80,8 +81,22 @@ def __init__(self, config, files):
             self.region_var = None
 
         # Cleanup
-        self.snapping = config["cleanup"]["snapping_threshold"]
-        self.segmentation_length = config["cleanup"]["segmentation_length"]
+        self._cleanup_options = config.get("cleanup", {})
+        if bool(self._cleanup_options) and any(
+            [v for k, v in self._cleanup_options.items()]
+        ):
+            self._cleanup_options.update({"cleanup": True})
+        # TODO: remove the attributes once cleanup function is used
+        self.cleanup = self._cleanup_options.get("cleanup", False)
+        self.snapping = self._cleanup_options.get("snapping_threshold", None)
+        self.segmentation_length = self._cleanup_options.get(
+            "segmentation_length", None
+        )
+        self.merge_lines = self._cleanup_options.get("merge_lines", None)
+        self.merge_on_id = self._cleanup_options.get("merge_on_id", None)
+        self.cut_at_intersections = self._cleanup_options.get(
+            "cut_at_intersections", None
+        )
```

Comment:
You can crete a private method 'set_cleanup_options', eventually these assignments will became outdated as the `@dataclass` type will take care of it with their default values.